### DISCUSSION TITLE
Header depending on page number

### DISCUSSION THREAD

**Discussion by KillTheMule** [typst/typst#2065] [General]
I'd like to set the header depending on page number. I'd like to do so programatically, i.e. not resort to using `#set page` manually. Here's my last recent try:

```typst
#set page(
  header: [
  #let x =  locate(loc => {
        counter(page).at(loc).first()
            })
  #if [#x] == [2] [stuffsies]
  ],
)
```
The problem is that `#x` is content on which I seem to be unable to match, although `[#x]` shows up in the document as `[2]`. Does anyone know how to do this? Thanks for any pointers :)

**Comment by Enivex** [comment]
Wrap the entire header in locate and do your logic inside there.

**Reply by KillTheMule** [reply]
That works, thanks!

**Comment by freundTech** [comment]
`locate` can only return content. The correct solution would be to put the `if` inside the locate and only return the finished content from it:
```typst
#set page(
  header: [
      #locate(loc => {
        let x = counter(page).at(loc).first()
        if x == 2 [stuffsies]
      })
  ]
)
```

You can shorten this a bit to 
```typst
#set page(
  header: locate(loc => {
      let x = counter(page).at(loc).first()
      if x == 2 [stuffsies]
  })
)
```

**Reply by KillTheMule** [reply]
Worked like a charm, thanks!

**Reply by DawnEver** [reply]
In version 0.12.0, `locate`  is deprecated now.Any alternative methods?
documentation link https://typst.app/docs/reference/introspection/locate

**Reply by laurmaedje** [reply]
The 0.12 version would look like this:
```typ
#set page(
  header: context {
      let x = counter(page).get().first()
      if x == 2 [stuffsies]
  })
)
```
See here: https://typst.app/docs/reference/context/
